### PR TITLE
ticket 6 and errors handled

### PR DESCRIPTION
**Pull request by HannahBennett101**

### CODE REVIEW

File: __tests__/app.tests.js [hunk 1]
```diff
@@ -106,4 +106,46 @@ describe('GET /api/articles/:article_id', () => {
...
+          comment_id: expect.any(Number),
+          votes: expect.any(Number),
+          created_at: expect.any(String),
+          author: expect.any(String),
+          body: expect.any(String)
+
+      })
+    });
+    expect(articleOneComments).toBeSortedBy("created_at", {descending: true, coerce: true, })
+  });
```
Comment (xViolaine) [inline __tests__/app.tests.js:132]: Another test where we expect a 200 when the ID is valid but has no comments -> responds with an empty array of comments

File: models/models.js [hunk 2]
```diff
@@ -18,9 +18,15 @@ exports.selectArticles = () => {
...
-};
\ No newline at end of file
+};
+
+exports.selectCommentsByArticleID = (article_id) => {
+    return db
+    .query(`SELECT article_id, comment_id, votes, created_at, author, body FROM comments WHERE article_id = $1 ORDER BY created_at DESC;`, [article_id])
+    .then(({rows}) => rows[0] === undefined ? Promise.reject({status:404, msg:"Article not found"}) : rows)
+    .catch(err => Promise.reject(err))
+}
```
Comment (xViolaine) [inline models/models.js]: Since we want to give a 200 when the comments are empty, but the article exists, we need to make sure to check the reason for why the query would be returning empty rows
- article exists but has no comments
- article does not exist 
are the two outcomes. We need to check whether the article exists in order to determine this, so this needs some change!

You can also check the comments first, and if those are empty, then do a check on whether the article exists. The order is up to you!

**Review by xViolaine — CHANGES_REQUESTED** [review]
